contracts/merger/TribeRagequit.sol

```diff
@@ -0,0 +1,106 @@
+//SPDX-License-Identifier: MIT
+pragma solidity ^0.8.4;
+
+import "./MergerBase.sol";
+import "../oracle/collateralization/ICollateralizationOracle.sol";
+import "../token/IFei.sol";
+import "@openzeppelin/contracts/utils/cryptography/MerkleProof.sol";
+import "@openzeppelin/contracts/utils/math/SafeCast.sol";
+
+/// @title Contract to exchange TRIBE with FEI post-merger
+/// @author elee
+contract TRIBERagequit is MergerBase {
+    using SafeCast for *;
+    using SafeERC20 for IERC20;
+
+    address public constant coreAddress = 0x8d5ED43dCa8C2F7dFB20CF7b53CC7E593635d7b9;
+    address public constant guardian = 0xB8f482539F2d3Ae2C9ea6076894df36D1f632775;
+
+    uint256 public intrinsicValueExchangeRateBase;
+
+    address public constant rewardsDripper = 0x3Fe0EAD3500e767F0F8bC2d3B5AF7755B1b21A6a;
+    IFei public constant fei = IFei(0x956F47F50A910163D8BF957Cf5846D573E7f87CA);
+
+    uint256 public immutable rageQuitEnd;
+
+    mapping(address => uint256) public claimed;
+
+    event Exchange(address indexed from, uint256 amountIn, uint256 amountOut);
+
+    bytes32 public merkleRoot;
+
+    constructor(
+        bytes32 root, 
+        uint256 _rageQuitEnd,
+        address tribeRariDAO
+    ) MergerBase(tribeRariDAO) {
+        merkleRoot = root;
+
+        rageQuitEnd = _rageQuitEnd;
+    }
+
+    /// @notice ragequit held TRIBE with FEI
+    /// @dev not gonna make it
+    /// @param amount the amount to redeem in TRIBE
+    /// @param key the amount of TRIBE allocated to the caller in the merkle drop
+    /// @param merkleProof a proof proving that the caller may redeem up to `key` amount of tribe
+    function ngmi(
+        uint256 amount,
+        uint256 key,
```

Comment:
this variable name should be changed to `tribeAllocAmount`, key doesn't make sense.

Reply:
Its the key to the merkle tree, this was the name Eddy used and I like it because it makes sure it isn't confused with some other quantity

Reply:
This naming still doesn't make sense. The actual key is `keccak256(abi.encodePacked(userAddress, RGTSnapshotAmount))`.

Reply:
renamed to totalMerkleAmount